src/screens/General/Dashboard.js

```diff
@@ -59,212 +55,207 @@ class Dashboard extends Component {
-				<View style = { mainContentStyle }>
-					<View style = {{ flex: 1, flexDirection: "row", justifyContent: "space-evenly" }}>
-						<ScrollView style = {{ flex: 1 }}>
-							<View style = {{ flex: 1, height: dimension.height * 1.3 }}>
-								{ this.renderHeader() }
-								<View style = {{ flex: 1, paddingLeft: "5%", paddingRight: "5%" }}>
-									<View style = {{ flex: 0.13, flexDirection: "row" }}></View>
-									<View style = {{ alignItems: "center", flex: 0.2, justifyContent: "center" }}>
-										<Text style = { [title, textColor] }>Upcoming Events</Text>
-									</View>
-									{ this.getFormattedEventList() }
-									<View style = {{ flex: 0.03, flexDirection: "row" }}></View>
-									<View style = {{ flexDirection: "row", alignItems: "flex-start", flex: 0.9, borderColor: "white" }}>
-										{ this.renderLeaderboard() }
-										<View style = {{ flex: 0.05 }}></View>
-										{ this.renderCommitteePanel() }
-									</View>
-								</View>
-								{ this.renderButtonLinks() }
+				<ScrollView>
+					<View style = { innerScrollContainer }>
+						{ this.renderHeader() }
+						<View style = { dashboardContent }>
+							<View style = { upcomingEventsContainter }>
+								<Text style = { [title, textColor] }>Upcoming Events</Text>
 							</View>
-						</ScrollView>
-						{ this.renderFlags() }
+							{ this.getFormattedEventList() }
+							<View style = { dashCommitteesContainer }>
+								{ this.renderLeaderboard() }
+								{ this.renderCommitteePanel() }
+							</View>
+						</View>
+						{ this.renderButtonLinks() }
 					</View>
-					{ this.renderNewFlag() }
-				</View>
+				</ScrollView>
+				{ this.renderFlags() }
+				{ this.rendercustomFlag() }
 			</SafeAreaView>
 		);
 	}
 
-	renderColorPicker() {
+	renderHeader() {
+		const {
+			headerContainer,
+			headerOptionsContainer
+		} = styles;
+
+		let dashColor = { backgroundColor: this.props.dashColor };
+		let chevronColor = { color: "white" };
+
 		return (
-			<Modal visible = { this.state.colorPicker } transparent = { true }>
-				<View style = { [styles.modalBackground, { backgroundColor: "transparent" }] }>
-					<ColorPicker
-						defaultColor = "#21252b"
-						oldColor = { this.props.dashColor }
-						onColorSelected = { color => this.colorPicked(color) }
-						style = { [styles.modalContent, { backgroundColor: "black" }] }
+			<View style = { [headerContainer, dashColor ] }>
+				{ this.renderGreeting() }
+				<View style = { headerOptionsContainer }>
+					{ this.renderHeaderFlag() }
+					<FontAwesomeIcon
+						style = { chevronColor }
+						name = "chevron-down"
+						onPress = { () => this.setState({ colorPickerVisible: true }) }
+						size = { 15 }
 					/>
 				</View>
-			</Modal>
+			</View>
 		);
 	}
 
-	renderHeader() {
+	renderGreeting() {
 		const {
-			greetingContainerStyle
+			textColor,
+			greetingContainer
 		} = styles;
 
+		const months = ["January", "February", "March", "April", "May", "June", "July",
```

Comment:
I think this information is saved somewhere else already when it's used in events. This information should be imported from somewhere and named something like monthsAsStrings.